### ISSUE TITLE
Manifest releaser cannot find lateset release for Release PR

### ISSUE THREAD

**Issue by luyuqing** [googleapis/release-please#1574]
Manifest releaser works successfully looking up latest release for regular PRs, and collects the right amount of commits. However when merging a Release PR created by release-please bot, it cannot find a release and trace too far back with a lot of commits.

here is my public test repo and actions result:  https://github.com/luyuqing/testWebApi

release.yml (Ignore the 2nd job build-release as it is not related with this issue)

```
on:
  push:
    branches:
      - main
name: release

jobs:
  release-please:
    runs-on: windows-latest
    steps:
      - uses: google-github-actions/release-please-action@v3
        with:
            token: ${{ secrets.GITHUB_TOKEN }}
            command: manifest

      - name: "echo push commits"
        run: echo ${{github.event.commits[0].author.name}}

  build-release:
    if: ${{startswith(github.event.commits[0].author.name, 'github-actions')}}
    runs-on: windows-latest
    needs: release-please
    steps:
      - uses: actions/checkout@v2.2.0
        with:
          fetch-depth: 0 
      - name: "Get tag"
        id: previoustag
        uses: "WyriHaximus/github-action-get-previous-tag@v1"

      - name: "just echo tag"
        run: echo ${{steps.previoustag.outputs.tag}}
``` 

release-please-config.json (Ignore the extra files config because that's just me playing around with extra-files, not related with this issue.)

```{
    "packages": {
        ".": {
            "release-type": "simple",
            "extra-files": [
                {
                    "type": "json",
                    "path": "Client/ClientApp/package.json",
                    "jsonpath": "version"
                },
                {
                    "type": "json",
                    "path": "Client/ClientApp/package-lock.json",
                    "jsonpath": "version"
                },
                {
                    "type": "json",
                    "path": "Client/ClientApp/package-lock.json",
                    "jsonpath": "$.packages[''].version"
                }
            ]
        }
    }
}
```

Actions output for a regular PR with "fix" commit:
```
Run google-github-actions/release-please-action@v3
√ Building pull requests
√ Building strategies by path
√ Collecting release commit SHAs
√ Collecting commits since all latest releases
√ Splitting 1 commits by path
√ Building candidate release pull request for path: .
√ Considering: 1 commits
√ Looking for open release pull requests
√ found 0 open release pull requests.
√ Looking for snoozed release pull requests
√ found 0 snoozed release pull requests.
‼ file version.txt did not exist
√ Starting GitHub PR workflow...
√ Successfully found branch HEAD sha "94ce379b4c753690bda440f3bda7efc11cfba4de".
√ Successfully created branch at https://api.github.com/repos/luyuqing/testWebApi/git/refs/heads/release-please--branches--main
√ Got the latest commit tree
√ Successfully created a tree with the desired changes with SHA 209c873b5aa1f5928c9f34[14](https://github.com/luyuqing/testWebApi/runs/7837255786?check_suite_focus=true#step:2:15)9fdf4368e9affa4d
√ Successfully created commit. See commit at https://api.github.com/repos/luyuqing/testWebApi/git/commits/af8eaf0e021df[18](https://github.com/luyuqing/testWebApi/runs/7837255786?check_suite_focus=true#step:2:19)a480676322f681e7e5efafcd0
√ Updating reference heads/release-please--branches--main to af8eaf0e0[21](https://github.com/luyuqing/testWebApi/runs/7837255786?check_suite_focus=true#step:2:22)df18a4806763[22](https://github.com/luyuqing/testWebApi/runs/7837255786?check_suite_focus=true#step:2:23)f681e7e5efafcd0
√ Successfully updated reference release-please--branches--main to af8eaf0e021df18a480676322f681e7e5efafcd0
√ Successfully opened pull request available at url: https://api.github.com/repos/luyuqing/testWebApi/pulls/100.
√ Successfully opened pull request: 100.
√ Successfully added labels autorelease: pending to issue: 100
```

Actions output when merging a Release PR by release-please bot, after the above "fix" PR:
```
Run google-github-actions/release-please-action@v3
√ Building pull requests
√ Building strategies by path
√ Collecting release commit SHAs
‼ Expected 1 releases, only found 0
‼ Missing 1 paths: .
‼ Expected 1 releases, only found 0
√ Collecting commits since all latest releases
√ Splitting 151 commits by path
√ No latest release found for path: ., component: , but a previous version (1.2.26) was specified in the manifest.
‼ No latest release pull request found.
√ Building candidate release pull request for path: .
√ Considering: [13](https://github.com/luyuqing/testWebApi/runs/7837293145?check_suite_focus=true#step:2:14)7 commits
‼ There are untagged, merged release PRs outstanding - aborting
√ Building releases
√ Building strategies by path
√ Building release for path: .
√ Creating 1 releases for pull #100
```

When release-please is trying to find the latest release to collect commits, does release-please differentiate between regular PRs and Release PRs?

**Comment by fastlorenzo** [comment]
I have the same issue when using it to release Helm chart (repo: https://github.com/fastlorenzo/helm-charts-1).
It is looking for the tag for the version to be released instead of the latest release.
For example, this PR to release v 1.0.0-beta.31 of the mailu Helm Chart: https://github.com/fastlorenzo/helm-charts-1/pull/64/files
We can see that the versions are correctly updated, however when I merge it I get the following:

```

Run google-github-actions/release-please-action@v3
❯ Fetching release-please-config.json from branch master
❯ Fetching .release-please-manifest.json from branch master
✔ Building pull requests
✔ Building strategies by path
❯ mailu: helm
✔ Collecting release commit SHAs
❯ release search depth: 400
❯ Fetching releases with cursor undefined
❯ Fetching releases with cursor Y3Vyc29yOnYyOpK0MjAyMi0xMS0wNVQwNjoxNjoyOVrOBOWNbQ==
⚠ Expected 1 releases, only found 0
⚠ Missing 1 paths: mailu
⚠ Expected 1 releases, only found 0
❯ looking for tagName: mailu-1.0.0-beta.31
✔ Collecting commits since all latest releases
❯ commit search depth: 500
❯ Set(0) {}
❯ Fetching merge commits on branch master with cursor: undefined
❯ Backfilling file list for commit: 4f72200e52e68f2cd88f70e51a6a9b6161af4c9b
❯ Found 3 files
<MAKE AN INSANE AMOUNT OF API CALLS, THEN RATE LIMITED>
```

**Comment by jimeh** [comment]
I've managed to work around the API rate limit issues by having a CI job that automatically updates the `last-release-sha` configuration option. Despite it not finding the correct tag, the backfill is limited to the SHA specified.

In the below snippet, the `post-release` job is the hacky workaround:

```yaml
  release-please:
    runs-on: ubuntu-latest
    needs: [test]
    if: github.ref == 'refs/heads/main'
    outputs:
      release_created: ${{ steps.release-please.outputs.release_created }}
      tag_name: ${{ steps.release-please.outputs.tag_name }}
      version: ${{ steps.release-please.outputs.version }}
      all: ${{ toJSON(steps.release-please.outputs) }}
    steps:
      - uses: google-github-actions/release-please-action@v3
        id: release-please
        with:
          command: manifest

  release:
    runs-on: ubuntu-latest
    needs: [release-please]
    if: needs.release-please.outputs.release_created
    steps:
      - name: do release stuff...
      # [...]

  post-release:
    runs-on: ubuntu-latest
    needs: [release-please]
    if: needs.release-please.outputs.release_created
    steps:
      - uses: actions/checkout@v3
        with:
          ref: "${{ github.ref }}"
      - name: update release-please's last-commit-sha
        run: |-
          jq '.["last-release-sha"] = "${{ github.sha }}"' \
            release-please-config.json > tmp && \
          mv tmp release-please-config.json
      - name: commit and push changes
        uses: stefanzweifel/git-auto-commit-action@v4
        with:
          commit_message: "ci(release-please): update last-release-sha"
          commit_user_name: github-actions[bot]
          commit_user_email: github-actions[bot]@users.noreply.github.com
          commit_author: >-
            github-actions[bot] <github-actions[bot]@users.noreply.github.com>
          skip_dirty_check: false
```

**Comment by chingor13** [comment]
Specifically for the bug about the GitHub action run immediately after merging the release PR -- that was fixed in https://github.com/google-github-actions/release-please-action/pull/753 (tag releases before trying to scan and create new PRs) and released in the v3.7.8 of the action. Looking at the original bug, this seems to be the same case as https://github.com/google-github-actions/release-please-action/issues/747

**Comment by jimeh** [comment]
@chingor13 Oh, amazing. I've confirmed that my workaround is no longer needed :)

It looks like v3.7.8 of the action was released a couple of hours before I first ran into this issue and came up with the workaround... lol
